backend/tournaments/models.py

```diff
@@ -0,0 +1,139 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+
+class Member(models.Model):
+  user = models.OneToOneField(User, primary_key=True,
+    help_text='Basic account information, username, password, email, etc.')
+
+  country = models.CharField(max_length=2, null=True, blank=True,
+    help_text='Two letter country code of the member. Lowercase.')
+
+  suspended_until = models.DateField(null=True, blank=True,
+    help_text='Until what date the user is suspended from games.')
+
+  reliability = models.IntegerField(help_text='Reliability rating.')
+
+  preferred_hours = models.CharField(max_length=24, default='y' * 24, help_text=
+    '24-character string of format "yyyyyrrrrwwwwwwwwyyyyyyy", UTC-based.')
+
+  preferred_control = models.CharField(default='45 45', max_length=100,
+    help_text='Comma separated list of ‘MM SS’ time controls.')
+
+  # TODO(crem): Do we need `status` here?
+
+
+class Tournament(models.Model):
+  short_name = models.SlugField(max_length=32,
+    help_text='E.g. monthly15, sb4, etc.')
+
+  name = models.CharField(max_length=160,
+    help_text='Snail Bucket Monthly 2015')
+
+  type_code = models.CharField(max_length=32,
+    help_text='Tournament type code, e.g. "RR", "Swiss", etc.')
+
+  signup_start = models.DateTimeField(help_text='When registration is open.')
+
+  signup_end = models.DateTimeField(
+    help_text='When registration is no longer open.')
+
+  games_start = models.DateTimeField(
+    help_text='Round 1 won\'t start before that time')
+
+  rounds_count = models.IntegerField(
+    help_text='Number of rounds in the tournament.')
+
+  rounds_start_cron = models.CharField(max_length=32,
+    help_text='Starting date of every round, in crontab format.')
+
+  min_bucket_size = models.IntegerField(
+    help_text='Minimum number of players in a bucket.')
+
+  max_bucket_size = models.IntegerField(
+    help_text='Maximum number of players in a bucket.')
+
+
+class Round(models.Model):
+  tournament = models.ForeignKey(Tournament)
+
+  start = models.DateTimeField(help_text='When the round starts.')
+
+
+class Bucket(models.Model):
+  tournament = models.ForeignKey(Tournament)
+
+  name = models.CharField(max_length=64, help_text='Name of the bucket.')
+
+  tds = models.ManyToManyField(Member)
+
+
+class BucketPlayer(models.Model):
+  bucket = models.ForeignKey(Bucket, related_name='players',
+    help_text='Player of which bucket.')
+
+  member = models.ForeignKey(Member, help_text='Which user.')
+
+  # TODO(crem): Do we need `status` here?
```

Comment:
Not necessarily. If someone withdraws or is banned, we can immediately set all his games in his future rounds.